app/views/supplementary/declaration_additional_actors.scala.html

```diff
@@ -16,32 +16,55 @@
 
 @import config.AppConfig
 @import controllers.supplementary.routes._
+@import controllers.util.Remove
 @import forms.supplementary.DeclarationAdditionalActors
 @import forms.supplementary.DeclarationAdditionalActors.PartyType._
 @import uk.gov.hmrc.play.views.html._
 @import utils.RadioOption
 
-@(appConfig: AppConfig, form: Form[DeclarationAdditionalActors])(implicit request: Request[_], messages: Messages)
+@import models.declaration.supplementary.DeclarationAdditionalActorsData
+@(appConfig: AppConfig, form: Form[DeclarationAdditionalActors], actors: Seq[DeclarationAdditionalActors])(implicit request: Request[_], messages: Messages)
 
 @main_template(
     title = messages("supplementary.additionalActors.title"),
     appConfig = appConfig
 ) {
-    @helpers.form(DeclarationAdditionalActorsController.saveAdditionalActors(), 'autoComplete -> "off") {
+    @helpers.form(DeclarationAdditionalActorsController.saveForm(), 'autoComplete -> "off") {
         @components.back_link("/customs-declare-exports/declaration/supplementary/representative-details")
 
+        @if(actors.nonEmpty) {
```

Comment:
Can we have some kind of a generic component for the table? It would be handy later. There is something similar: `single_value_elements_table`, but it won't handle case when there is more fields.

Reply:
Actually that would be a good idea, but also I have no idea how to do it and how can it be generic.

Reply:
I think we need to address this across the service once we have the full journey.